geopyspark/geotrellis/protobufcodecs.py

```diff
@@ -273,6 +273,42 @@ def create_partial_tuple_decoder(key_type):
 
     return partial(tuple_decoder, key_decoder=key_type)
 
+def image_rdd_decoder(proto_bytes, key_decoder):
+    """Decodes tuple of ``(K, bytes)`` where the bytes are the PNG bytes of the raster and
+    the ``K`` is the raster's corresponding key.
+
+    Args:
+        proto_bytes (bytes): The ProtoBuf encoded bytes of the ProtoBuf class.
+        key_decoder (str): The name of the key type of the tuple.
+
+    Returns:
+        tuple
+    """
+
+    tup = tupleMessages_pb2.ProtoTuple.FromString(proto_bytes)
+    image_bytes = tup.imageBytes
+
+    if key_decoder == "ProjectedExtent":
+        return (from_pb_projected_extent(tup.projectedExtent), image_bytes)
+    elif key_decoder == "TemporalProjectedExtent":
+        return (from_pb_temporal_projected_extent(tup.temporalProjectedExtent), image_bytes)
+    elif key_decoder == "SpatialKey":
+        return (from_pb_spatial_key(tup.spatialKey), image_bytes)
+    else:
+        return (from_pb_space_time_key(tup.spaceTimeKey), image_bytes)
+
+def create_partial_image_rdd_decoder(key_type):
```

Comment:
The reason I chose that name is because right now, it just deals with PNG and soon GeoTiff bytes. However, I can change this if it's not clear or confusing.